src/xdocs/index.xml.vm

```diff
@@ -217,6 +217,19 @@
       </div>
     </section>
 
+    <section name="Supported Java Language Version">
+      <p>
+        Checkstyle can parse all Java language features introduced
+        in <a href="https://docs.oracle.com/javase/specs/jls/se21/html/index.html">Java 21</a>
+        and below. We may support preview features depending on community demand,
```

Comment:
What about 22? Why did we stop at 21?

From our doc, unnamed patterns/variables were in 22.

Reply:
done to 22.